src/coreclr/jit/llvm_types.h

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*
+ * Types used in LLVM compilation that are abstractions of the VM types
+*/
+
+#ifndef _LLVM_TYPES_H_
+#define _LLVM_TYPES_H_
+
+#include "corinfo.h"
+
+struct TypeDescriptor
+{
+    size_t                fieldCount;
+    CORINFO_FIELD_HANDLE* fields;
+};
+
+struct FieldDesc
+{
+private:
+    int                  _fieldOffset;
+    CORINFO_CLASS_HANDLE _classHandle;
+    CorInfoType          _corType;
+
+public:
+    FieldDesc()
+    {
+    }
+
+    int getFieldOffset()
+    {
+        return _fieldOffset;
+    }
+
+    CORINFO_CLASS_HANDLE getClassHandle()
+    {
+        return _classHandle;
+    }
+
+    CorInfoType getCorType()
+    {
+        return _corType;
+    }
+
+    void setFieldData(int fieldOffset, CorInfoType corType, CORINFO_CLASS_HANDLE classHandle)
```

Comment:
It would be cleaner to set these in the constructor and make the type fully immutable.

Reply:
Yes, I did think about that but as the structs are created `new StructDesc(fieldCount, new FieldDesc[fieldCount]);`  I didn't know how what would be the right pattern to avoid reassigning the `FieldDesc` in each array index.

Reply:
I think it's fine to re-assign them, it is what `setFieldData` already effectively does.
```c
FieldDesc* fields = new FieldDesc[fieldCount]; // Will require a parameterless constructor

for (...) { fields[i] = StructDesc(offset, type, handle); }

StructDesc* structDesc = new StructDesc(fieldCount, fields);
```

Reply:
Yes, of course.